user/views.py

```diff
@@ -1,17 +1,100 @@
+        if request.method == 'POST':
+            email = request.POST.get('email')
+            if not email:
+                messages.warning(request, "Email required")
+                return redirect('index')
+            if User.objects.filter(email=email):
+                messages.warning(request, "User already exists with this email")
+                return redirect('index')
 
+            full_name = request.POST.get('name')
+            full_username = "".join(full_name.split())
+            user_name = full_username
 
-def home(request):
-    if request.user.is_authenticated:        
-        return render(request,'tweet/home.html')
-    return redirect('index')
\ No newline at end of file
+            while User.objects.filter(username=user_name).exists():
+                random_number = random.randint(1, 9999)
+                user_name = f"{full_username}{random_number}"
+            User.objects.create(
+                password = make_password(request.POST.get('password')),
+                username = user_name,
+                first_name = full_name,
+                email=email
+            )
+            messages.success(request, "Login to continue")
+            return redirect('index')
+
+
+
+
+
+def signin(request):
+    if request.method == 'POST':
+        unique_identity = request.POST.get('email')
+        password = request.POST.get('password')
+        select_user = User.objects.filter(Q(email=unique_identity) | Q(phone=unique_identity) | Q(username=unique_identity))
+        if select_user:
+            unique_identity = select_user.first().username
+            user = authenticate(request,username=unique_identity,password=password)
+            login(request, user) if user else messages.error(request, "Incorrect password!!!")
+        else:
+            messages.warning(request, "Account not found")
+    return redirect('index')
+
+    # if request.user.is_authenticated:
+    #     return redirect('index')
+    # else:
+    #     if request.method == 'GET':
+    #         return redirect('index')
+
+    #     if request.method == 'POST':
+    #         unique_identity = request.POST.get('email')
+    #         password = request.POST.get('password')
+
+    #         if User.objects.filter(Q(email=unique_identity) | Q(phone=unique_identity) | Q(username=unique_identity)):
+    #             unique_identity = User.objects.filter(Q(email=unique_identity) | Q(phone=unique_identity) | Q(username=unique_identity)).first().username
+    #             user = authenticate(
+    #                     request,
+    #                     username=unique_identity,
+    #                     password=password
+    #                 )
+    #             if user:
+    #                 login(request, user)
+    #             else:
+    #                 messages.error(request, "Incorrect password!!!")
+    #         else:
+    #             messages.warning(request, "Account not found")
+    #     return redirect('index')
```

Comment:
don't leave commented code

Reply:
just checking all functions are working correctly and if you decline the new code then the previous one is there.